### PR TITLE
docs(root): add guidance and code examples for min and max props

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes

Add guidance and code examples for min and max props on text field. You'll need to run npm pack for web-components, react, and docs on develop in the ui kit and install those locally to see the example working.

## Related issue

#427

## Checklist

- [x] I have [manually accessibility tested](https://design.sis.gov.uk/accessibility/testing/manual-testing) any changes, if relevant.
- [x] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

File: src/content/structured/components/text-input/guidance.mdx [hunk 1]
```diff
@@ -74,6 +74,8 @@ Use text fields as part of a form or other data collection page for capturing te
 
 Use clear labels to indicate what information should be entered into the field.
 
+When using min or max props, ensure to add visual indication of these values in the label or helper text.
```
Comment (DBD324) [inline src/content/structured/components/text-input/guidance.mdx:77]: "When setting minimum and maximum values using the `min` and `max` props, make sure to add an indication of these formatting rules in the helper text."

Comment (DBD324) [reply src/content/structured/components/text-input/guidance.mdx:77]: This is actually better palced in the next section for 'helper text'.

File: src/content/structured/components/text-input/guidance.mdx [hunk 2]
```diff
@@ -74,6 +74,8 @@ Use text fields as part of a form or other data collection page for capturing te
...
+
 ### Helper text
 
 Use helper text in addition to the field label when additional information helps users understand what information is required.
```
[the remark this answers is not in the document]
Comment (DBD324) [reply src/content/structured/components/text-input/guidance.mdx:81]: This should appear below the image.

File: src/content/structured/components/text-input/code.mdx [hunk 3]
```diff
@@ -283,6 +283,50 @@ export const snippetsValidation = [
   </div>
 </ComponentPreview>
 
+### With min and max
+
+In this example, the error message will change for inputted values below min and above max.
```
Comment (DBD324) [inline src/content/structured/components/text-input/code.mdx:288]: In this example, an error message will display when the input is either below the `min` value or above the `max` value. This only effects numeric input fields.

**Review by DBD324 — CHANGES_REQUESTED** [review]
A couple of specific changes requested on the edits in this file. There are a few other additions to the guidance that we would need to make that talks about how to add the min and max values, as well as how to change the text field to allow numeric input (as well as other allowable input types such as password)

File: src/content/structured/components/text-input/guidance.mdx [hunk 4]
```diff
@@ -87,6 +87,8 @@ Use helper text in addition to the field label when additional information helps
   />
 </DoubleDoDontCaution>
 
+When setting min and max values using the min and max props, make sure to add an indication of these formatting rules in the helper text.
```
Comment (DBD324) [inline src/content/structured/components/text-input/guidance.mdx:90]: "When setting minimum and maximum values..." and can we use the code styling for 'min' and 'max' props?

File: src/content/structured/components/text-input/code.mdx [hunk 5]
```diff
@@ -283,6 +283,53 @@ export const snippetsValidation = [
   </div>
 </ComponentPreview>
 
+### With min and max
```
Comment (DBD324) [inline src/content/structured/components/text-input/code.mdx:286]: "With min and max values"

**Review by DBD324 — CHANGES_REQUESTED** [review]

**Review by DBD324 — DISMISSED** [review]

File: src/content/structured/components/text-input/code.mdx [hunk 6]
```diff
@@ -325,6 +325,53 @@ export const snippetsValidation = [
...
+  min="1"
+  max="4"
+>
+</ic-text-field>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcTextField
+  validationStatus="error"
+  validationText="now it has really gone to (coffee) pot"
```
Comment (MI6-255) [inline src/content/structured/components/text-input/code.mdx:351]: I think these validations don't show what we want them to with min/max, I think it'd be more setting the value to say 6 (over the max value) and showing that validation?

**Review by MI6-255 — CHANGES_REQUESTED** [review]

File: src/content/structured/components/text-input/code.mdx (hunk 6 above)
Comment (MI6-255) [reply src/content/structured/components/text-input/code.mdx:352]: I think only the web component was updated here. But also, unless I'm getting the wrong end of the stick, I think the validation needs to just be removed? If you have the value set to 6 in this example, the validation is automatically generated. Which I think is what we want to display here?

**Review by MI6-255 — CHANGES_REQUESTED** [review]

**Review by DBD324 — APPROVED** [review]
Changes look good. Thank you

**Pull request merged**